### PR TITLE
Add optional params key_name and value_name to customize dict lookup plugin

### PR DESCRIPTION
**Pull request by jangel97**
Hello there, 

In my playbooks I am using the following data `{{ lookup('dict', ansible_devices) }}` to filter and query the devices in my systems.

It would be cool for me if I could use something like this `{{ lookup('dict', ansible_devices, key_name='device', value_name='device_info') }}`, it would help me improve my playbook readability.

It would make sense to me having optional parameters to customize the behaviour of the key and the value created by the lookup plugin dict.

Let me know your thoughts.

Kind regards,

### CODE REVIEW

File: lib/ansible/plugins/lookup/dict.py [hunk 1]
```diff
@@ -63,6 +77,9 @@ class LookupModule(LookupBase):
 
     def run(self, terms, variables=None, **kwargs):
 
+        key_name = kwargs.get('key_name', 'key')
+        value_name = kwargs.get('value_name', 'value')
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/dict.py]: you should not use kwargs directly but 'get_option' in plugins

File: lib/ansible/plugins/lookup/dict.py [hunk 2]
```diff
@@ -10,12 +10,20 @@
...
                 - A list of dictionaries
             required: True
+        key_name:
+            description:
+                - Use it to override the 'key' key name in the dictionaries. By default it will be set to 'key'.
+            required: False
+        value_name:
+            description:
+                - Use it to override the 'key' key name in the dictionaries. By default it will be set to 'value'.
+            required: False
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/dict.py:22]: missing, `version_added` and `default`, which also means you don't need to state it in the description.

File: lib/ansible/plugins/lookup/dict.py (hunk 1 above)
Comment (jangel97) [reply lib/ansible/plugins/lookup/dict.py]: hi, I just added the `self.get_option` to the lookup plugin. Tested in my local and it grabs the value correctly. Please, let me know if you find have any issues.
Thank you!

File: lib/ansible/plugins/lookup/dict.py (hunk 2 above)
Comment (jangel97) [reply lib/ansible/plugins/lookup/dict.py:22]: I added `default`, so get_options defaults to `key` and `value`. Do I need to add `version_added` as well? Or would you prefer that I remove `description`?
Thank you!

**Comment by bcoca** [comment]
Hi!

Thanks very much for your submission to Ansible.  It means a lot to us that you've taken time to contribute.

Unfortunately, we're not sure if we want this feature in the program, and I don't want this to seem confrontational.
Our reasons for this are:

During review it was pointed out that the `dict2items` filter already supports this and that this lookup itself is something we are considering for deprecation/removal (it is really a filter in lookup form, transforms data but does not query it from a source).


However, we're absolutely always up for discussion.
Because this project is very active, we're unlikely to see comments made on closed tickets and we lock them after some time.
If you or anyone else has any further questions, please let us know by using any of the communication methods listed in the page below:

   * https://docs.ansible.com/ansible/latest/community/communication.html

In the future, sometimes starting a discussion on the development list prior to implementing a feature can make getting things included a little easier, but it's not always necessary.

Thank you once again for this and your interest in Ansible!

**Pull request closed without merging**